### PR TITLE
Enable formatting with ocamlformat

### PR DESCRIPTION
**Pull request by NathanReb**
I'm opening this as draft to make sure `ocaml-ci` checks it right and I'd like to merge some of @gpetiot PRs first so we spare him some rebase nightmares.

### CODE REVIEW

File: bin/bistro.ml [hunk 1]
```diff
@@ -6,56 +6,57 @@
...
+        let args = add args "--pkg-version" version in
+        let args = if keep_v then Cmd.(args % "--keep-v") else args in
+        let args = add args "--tag" tag in
+        let args = add_l args "--pkg-names" pkg_names in
+        let dune_release = Cmd.(v "dune-release") in
+        OS.Cmd.run Cmd.(dune_release % "distrib" %% args) >>= fun () ->
+        OS.Cmd.run Cmd.(dune_release % "publish" %% args) >>= fun () ->
+        OS.Cmd.run Cmd.(dune_release % "opam" %% args % "pkg") >>= fun () ->
+        OS.Cmd.run Cmd.(dune_release % "opam" %% args % "submit") >>= fun () -> Ok 0)
   |> Cli.handle_error
```
Comment (gpetiot) [inline bin/bistro.ml]: we should rewrite: `Cli.handle_herror (...) `

File: bin/lint.ml [hunk 2]
```diff
@@ -8,56 +8,60 @@ open Bos_setup
...
+        Pkg.infer_pkg_names Fpath.(v ".") pkg_names >>= fun pkg_names ->
+        let pkg = Pkg.v ~dry_run ?name ?version ~keep_v ?tag () in
+        OS.Dir.current () >>= fun dir ->
+        List.fold_left
+          (fun acc name ->
+            acc >>= fun acc ->
+            let pkg = Pkg.with_name pkg name in
+            Lint.lint_pkg ~dry_run ~dir pkg lints >>= fun n -> Ok (acc + n))
+          (Ok 0) pkg_names)
   |> Cli.handle_error
```
Comment (gpetiot) [inline bin/lint.ml]: same here

File: bin/lint.ml [hunk 3]
```diff
@@ -8,56 +8,60 @@ open Bos_setup
...
-  let test = [ "std-files", `Std_files;
-               "opam", `Opam; ]
-  in
-  let doc = strf "Test to perform. $(docv) must be one of %s. If unspecified
-                  all tests are performed." (Arg.doc_alts_enum test)
+  let test = [ ("std-files", `Std_files); ("opam", `Opam) ] in
+  let doc =
+    strf
+      "Test to perform. $(docv) must be one of %s. If unspecified\n\
+      \                  all tests are performed." (Arg.doc_alts_enum test)
```
Comment (gpetiot) [inline bin/lint.ml]: to reformat

File: bin/opam.ml [hunk 4]
```diff
@@ -299,88 +262,92 @@ let field =
...
+    \             extension, uses an existing file with the same path but a `.descr`\n\
+    \             extension. If the opam file name is `opam` uses a `descr`\n\
+    \             file in the same directory. If these files are not found\n\
+    \             a description is extracted from the the readme (see\n\
+    \             option $(b,--readme)) as follow: the first marked up\n\
+    \             section of the readme is extracted, its title is parsed\n\
+    \             according to the pattern '\\$(NAME) \\$(SEP) \\$(SYNOPSIS)',\n\
+    \             the body of the section is the long description. A few\n\
+    \             lines are filtered out: lines that start with either\n\
+    \             'Home page:', 'Contact:' or '%%VERSION'."
```
Comment (gpetiot) [inline bin/opam.ml]: to reformat

File: bin/opam.ml [hunk 5]
```diff
@@ -299,88 +262,92 @@ let field =
...
+        \         The action needs a distribution archive to operate, see\n\
+        \         dune-release-distrib(1) or the $(b,--dist-file) option." );
+    `I
+      ( "$(b,submit)",
+        "submits a package created with the action $(b,pkg) the OCaml\n\
+        \         opam repository. This requires configuration to be created\n\
+        \         manually first, see $(i, dune-release help files) for more\n\
+        \         details." );
+    `I ("$(b,field) $(i,FIELD)", "outputs the field $(i,FIELD) of the package's opam file.");
+  ]
```
Comment (gpetiot) [inline bin/opam.ml]: to reformat

File: bin/tag.ml [hunk 6]
```diff
@@ -9,89 +9,95 @@ open Dune_release
...
   let doc = "Commit-ish $(docv) to tag." in
-  Arg.(value & opt string "HEAD" & info ["commit"] ~doc ~docv:"COMMIT-ISH")
+  Arg.(value & opt string "HEAD" & info [ "commit" ] ~doc ~docv:"COMMIT-ISH")
 
 let msg =
-  let doc = "Commit message for the tag. If absent, the message
-             'Distribution $(i,VERSION)' is used."
+  let doc =
+    "Commit message for the tag. If absent, the message\n\
+    \             'Distribution $(i,VERSION)' is used."
```
Comment (gpetiot) [inline bin/tag.ml]: to reformat

File: lib/lint.ml [hunk 7]
```diff
@@ -45,26 +40,25 @@ let lint_std_files ~dry_run pkg =
...
+        | `Ok ->
+            Logs.app (fun m ->
+                m "%a @[lint@ %s %a.@]" Text.Pp.status `Ok file_kind Text.Pp.path file);
+            errs
+        | `Fail msgs ->
+            Logs.app (fun m ->
+                m "%a @[<v>@[lint@ %s %a:@]@,@[%a messages:@]@,%a@]" Text.Pp.status `Fail file_kind
+                  Text.Pp.path file Cmd.pp cmd Fmt.lines msgs);
+            errs + 1)
   |> Logs.on_error_msg ~use:(fun () -> errs + 1)
```
Comment (gpetiot) [inline lib/lint.ml]: same here

File: .ocamlformat [hunk 8]
```diff
@@ -0,0 +1,3 @@
+version=0.12
+profile=conventional
+margin=100
```
Comment (samoht) [inline .ocamlformat]: why 100?

Comment (NathanReb) [reply .ocamlformat]: Personal preference, I'm happy to remove it if anybody objects!

Comment (gpetiot) [reply .ocamlformat]: I personally like to keep the default 80 col width in my editor, but I don't know how many people do that as well.

Comment (NathanReb) [reply .ocamlformat]: Let's stick to the default then!

**Comment by gpetiot** [comment]
You can merge now, I will rebase my other PRs.

**Pull request merged**